SportsTeamDetail/Modules/DetailModule/UseCases/GetTeamDetailsUseCase.swift

```
@@ -0,0 +1,95 @@
+import UIKit
+
+protocol GetTeamDetailsUseCaseProtocol {
+  func getTeam(completion: @escaping (Result<Team, Error>) -> Void)
+}
+
+final class GetTeamDetailsUseCase: GetTeamDetailsUseCaseProtocol {
+  func getTeam(completion: @escaping (Result<Team, Error>) -> Void) {
+    RepositoryFallbackStrategy().fetchData { result in
+      switch result {
```

Here I'm trying to simulate a production scenario that involves a fallback scenario. Since I'm using mock data, I gave the remote fetch an error as default. This way we can load all the mocked data from the local (mocked) repository